### PR TITLE
feat: update `eslint-plugin-eslint-plugin` + update configs

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Re-submission of https://github.com/mysticatea/eslint-plugin/pull/41

---

BREAKING CHANGE: `+eslint-plugin` config now has extra rules from `eslint-plugin-eslint-plugin`

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
@eslint-community/eslint-tsc @eslint-community/core-team @eslint-community/mysticatea-eslint-plugin Since [this plugin is adhering the same semantic versioning policy as ESLint](https://github.com/eslint-community/eslint-plugin-mysticatea#-semantic-versioning-policy), adding new rules to a plugin would be a breaking change, so this PR would need to be released in a major release I guess?

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "@typescript-eslint/eslint-plugin": "^2.34.0",
     "@typescript-eslint/parser": "^2.34.0",
     "eslint-plugin-eslint-comments": "^3.2.0",
-    "eslint-plugin-eslint-plugin": "^2.3.0",
+    "eslint-plugin-eslint-plugin": "^3.6.1",
```
Comment (ota-meshi) [inline package.json]: Is it possible to use version 4? https://github.com/eslint-community/eslint-plugin-eslint-plugin/blob/v4.4.1/package.json#L71-L73
Or is it possible to change the ESLint version that the plugin supports and use version 5?

**Review by ota-meshi — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (MichaelDeBoey) [reply package.json:19]: I think the supported Node version should be in line with what ESLint supports

Comment (MichaelDeBoey) [reply package.json:19]: `eslint-plugin-eslint-plugin` will be included when going to ESLint v8

I think the supported Node version should be in line with what ESLint supports

Comment (MichaelDeBoey) [reply package.json]: `eslint-plugin-eslint-plugin` will be included when going to ESLint v8

I also think the supported Node version should be in line with what ESLint supports

**Comment by mdjermanovic** [comment]
> adding new rules to a plugin would be a breaking change

I assume you mean adding new rules to configs exported from the plugin. I agree that should be a breaking change.

File: package.json (hunk 1 above)
Comment (ota-meshi) [reply package.json]: I believe `eslint-plugin-eslint-plugin` v4 supports ESLint v6. So I think we can use v4.
I don't think we should wait for ESLint v8.
https://github.com/eslint-community/eslint-plugin-eslint-plugin/blob/v4.4.1/package.json#L71-L73

**Comment by MichaelDeBoey** [comment]
> I assume you mean adding new rules to configs exported from the plugin

@mdjermanovic That's indeed what I meant 👍

@eslint-community/mysticatea-eslint-plugin I decided to make all the current outstanding PRs a breaking change & merge them into an `alpha` branch, so we can have 1 major release instead of one for every merged PR

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**